src/bidiMapper/utils/cdpClient.ts

```diff
@@ -13,50 +13,150 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { ServerBinding, AbstractServer } from './iServer';
 import { log } from './log';
 const logCdp = log('cdp');
 
-// TODO: Remove. This is a quick test to verify we can import a node module.
-import { version } from 'devtools-protocol/json/browser_protocol.json';
-logCdp(`Using CDP version: ${version}`);
+import { EventEmitter } from './events';
 
-export class CdpClient extends AbstractServer {
-  private _commandCallbacks: Map<number, (messageObj: any) => void> = new Map();
+import * as browserProtocol from 'devtools-protocol/json/browser_protocol.json';
+import * as jsProtocol from 'devtools-protocol/json/js_protocol.json';
+import ProtocolProxyApi from 'devtools-protocol/types/protocol-proxy-api';
 
-  constructor(cdpBinding: ServerBinding) {
-    super(cdpBinding);
+export type CdpClient = EventEmitter & ProtocolProxyApi.ProtocolApi;
```

Comment:
I like the approach with dynamic generating methods, but it can confuse IDE. Alternative I can think of is generating a TS file based on JSON. Not sure if we need it now though. WDYT?

Reply:
Right now, the IDE uses the pre-existing types from https://github.com/ChromeDevTools/devtools-protocol/tree/master/types. The nice thing about using pre-built type definitions is that all of these types are known at compile time. The drawback is there is no implementation and we need to generate the implementation at runtime. If there's a bug in how we generate the implementation, then the IDE's understanding of the types won't match reality. I think this code is correct though :)

If we generate our own typescript files that include an implementation, it will ensure the "compile-time" and "runtime" types always match, but we'll need an extra pre-build step to generate the files. The IDE won't recognize any CDP types until the developer builds at least once.

It might be useful to do this in the future if we need more control or want to have a different API than the one defined in `devtools-protocol`. For now though, I think dynamically generating the methods is good enough.